### ISSUE TITLE
docker-compose: Interactive mode is not yet supported on Windows.

### ISSUE THREAD

**Issue by reypm** [LaraDock/laradock#241]
I'm following the guide on the README for setup my local environment I am trying in Docker for Windows and I am using Windows 10. As you can see everything is up and running:
```
docker ps
CONTAINER ID        IMAGE                COMMAND                  CREATED             STATUS              PORTS                                      NAMES
dfd3c765e845        laradock_nginx       "nginx"                  7 seconds ago       Up 6 seconds        0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp   laradock_nginx_1
4e37aecdaa93        laradock_php-fpm     "php-fpm"                8 seconds ago       Up 7 seconds        9000/tcp                                   laradock_php-fpm_1
6e74f3b98251        laradock_redis       "docker-entrypoint.sh"   9 seconds ago       Up 7 seconds        0.0.0.0:6379->6379/tcp                     laradock_redis_1
3bb3c21a02ae        laradock_mysql       "docker-entrypoint.sh"   9 seconds ago       Up 7 seconds        0.0.0.0:3306->3306/tcp                     laradock_mysql_1
8d3338a1722b        laradock_workspace   "/sbin/my_init"          9 seconds ago       Up 8 seconds                                                   laradock_workspace_1
```
I am trying to run the following command but I get an error:
```
docker-compose exec workspace bash
[31mERROR[0m: Interactive mode is not yet supported on Windows.
Please pass the -d flag when using `docker-compose exec`.
```
Any advice for the ones running Docker on Windows?

**Comment by LarryEitel** [comment]
Are you using gitbash or vanilla CMD?

**Comment by LarryEitel** [comment]
I created a function and put it in `~/.aliases` which is loaded by `~/.bashrc`:
```
function dcx()
{
	DCX_CONTAINER=$(docker-compose ps | grep "$1" | cut -d" " -f 1)
	echo Bashing into: $DCX_CONTAINER
	winpty docker exec -it ${DCX_CONTAINER} bash
}
```
I have other aliases beginning with `dc` that stand for `docker-compose`. 
`dcx` means docker-compose EXECute. :)

Then I run:
$ `dcx workspace`

And it drops me into:
`root@c5e8eb4f0608:/var/www/laravel#`

**Comment by LarryEitel** [comment]
Actually as I look at this function, probably better to name it `dcbash`. :)

**Comment by reypm** [comment]
@LarryEitel I am using vanilla CMD although I've gitbash installed. Regarding your function where I should put this on Windows?

**Comment by LarryEitel** [comment]
~/.bashrc or ~/.profile

/c/Users/<name>/[.bashrc | .profile]

Try .bashrc first.

**Comment by reypm** [comment]
@LarryEitel I am missing something here:

```
D:\Sites\laradock>dcx workspace
'dcx' is not recognized as an internal or external command,
operable program or batch file.
```

What is that?

Also do I need to run the following command all the time for get the containers up?
`docker-compose up -d nginx mysql redis`

**Comment by reypm** [comment]
@LarryEitel nvm, I have moved from vanilla CMD to GitBash and it works. Now I am in `/var/www/laravel` but where this volume is mapped in the host? I mean in the windows PC?

**Comment by LarryEitel** [comment]
@reypm 

docker-compose.yml
```
### Laravel Application Code Container ######################

    volumes_source:
        build: ./volumes/application
        volumes:
            - ../:/var/www/laravel
```

This shows laravel is somewhere like: `/c/Users/laravel/laradock`.

I have sharing of `C` turned on in Docker settings. I put my stuff `/c/_dk/laravel/laradock`.

**Comment by reypm** [comment]
@LarryEitel perfect, I get it, now, lets said I want to put all my sources under `D:\Sites\laravel` do I need to modify something on the `docker-compose.yml`? If so ... what? Also I need to share my D drive in Windows from Docker config, right?

**Comment by LarryEitel** [comment]
Something like `/d/Sites/laravel`.

**Comment by mjcriswell** [comment]
I had the same problem in Powershell. I was able to enter a container `docker exec -it [container-id] bash`.

**Comment by Mahmoudz** [comment]
closed by #315
